### PR TITLE
docs(contributing): update existing docs to be inline with present ways of working

### PR DESCRIPTION
**Pull request by Parsium**
### Proposed behaviour

- Correct name for `DefinitionList`'s test story - so it does not appear at the bottom of the story list in storybook when run locally:
Screenshot 2022-10-19 at 12 02 09
- Rewrite existing docs on contributing to be inline with our current workflow - with the target audience being **developers who need concise details on how to get started**.
- Move details on legacy cypress E2E test to a separate file.
- Add new PR checklist item for checking documentation.
- In storybook docs, add `How to contribute` page and move `Development roadmap` page under new `Contributing` section. `How to contribute` provides a link to `CONTRIBUTING.md` on github.

### Current behaviour

- Test story for `DefinitionList` has inconsistent name - so it appears in a separate dropdown category in the story list when storybook is run locally:
Screenshot 2022-10-19 at 12 03 12

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- Double check for errors in grammar and ambiguous statements
- Suggest possible improvements to make docs clearer for new contributors

### CODE REVIEW

File: contributing/documentation-styleguide.md [hunk 1]
```diff
@@ -109,36 +113,50 @@ When writing a style guide, consider:
 Use hyphens to separate words in the file name and save it as a `.md` file in the relevant directory. If you need to add supporting images, save them in an `/images` directory within the same directory where you wish to store the style guide.
 
 #### Other Internal Documentation
```
Comment (edleeks87) [inline contributing/documentation-styleguide.md:115]: Are these headings at the correct hierarchy now you've updated the other ones?

image

File: contributing/writing-carbon-code.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+// If component should come with subcomponents - export all as named exports
+export { default as Select } from "./select.component";
+export { type SelectProps } from "./select.component";
+
+export { default as Option } from "./option/option.component";
+export { type OptionProps } from "./option/option.component";
+```
+
+> If you find any inconsistencies in module structure you are not obligated to change it. For example, if you need to work on that component to fix an unrelated bug. However, we are committed to provide a consistent interface for our components and will make changes where appropriate.
```
Comment (edleeks87) [inline contributing/writing-carbon-code.md:75]: May want to include that any changes will be appropriately semantically versioned

File: contributing/writing-carbon-code.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+export const PrimaryButton: ComponentStory<typeof Button> = () => (
+  <Button variant="primary">Click me!</Button>
+);
+
+export const SecondaryButton: ComponentStory<typeof Button> = () => (
+  <Button variant="secondary">Click me!</Button>
+);
+```
+
+For creating the component docs pages you see on [carbon.sage.com](https://carbon.sage.com), these should be written separately in the [MDX](https://storybook.js.org/docs/react/writing-docs/mdx) format as a `<component>.stories.mdx` file. This allows us to decouple our documentation from our story code. We then import our previously-written Typescript stories from `<component>.stories.tsx` into this new file, which Storybook uses to build our official docs for the component.
```
[the remark this answers is not in the document]
Comment (edleeks87) [reply contributing/writing-carbon-code.md:101]: It's probably okay to leave the hyphen in tbh, both seem correct 🤷‍♂️

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -119,16 +105,15 @@ Whichever workflow you use when you create the pull request you should include a
...
-
+- [Git commit messages](#git-commit-messages) follow the styleguide below.
+- The code is fully tested and has full coverage.
+- Cypress automation tests are included.
+- The code follows our existing coding conventions.
+- It has detailed commit messages and squashed commits. PRs with commit messages like 'WIP' and 'addressed comments', or long lists of commits will likely require amending."
+- The code does not introduce unnecessary dependencies (no jQuery!).
+- Storybook is updated to include examples.
+- Theme support is provided.
+- Typescript `d.ts` files are provided or updated.
```
Comment (edleeks87) [inline CONTRIBUTING.md:116]: `TypeScript`

File: contributing/writing-carbon-code.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+> If you find any inconsistencies in module structure you are not obligated to change it. For example, if you need to work on that component to fix an unrelated bug. However, we are committed to provide a consistent interface for our components and will make changes where appropriate.
+
+## Documenting with Storybook
+
+We use [Storybook](https://storybook.js.org/) for our documentation - which uses example *"stories"* to demo how our components can be used with certain props or how a particular layout can be achieved. Alongside demoing component behaviour, we also use stories for automated visual testing using [Chromatic](https://www.chromatic.com/), so we require stories to be written for all our components. 
+
+> If you are new to Storybook, we recommend first reading their [official docs](https://storybook.js.org/docs/react/get-started/introduction), particularly this section on [how to write stories](https://storybook.js.org/docs/react/writing-stories/introduction#how-to-write-stories), so you are familiar with the following patterns.
+
+New stories should be written in Typescript as a `<component>.stories.tsx` file to allow for type checking:
```
Comment (edleeks87) [inline contributing/writing-carbon-code.md:83]: `TypeScript`

File: contributing/documentation-styleguide.md (hunk 1 above)
Comment (Parsium) [reply contributing/documentation-styleguide.md:115]: Just gone back through the files and realised some of the levels were incorrect in `documentation-styleguide.md`. I will correct these 👍🏼

**Review by edleeks87 — DISMISSED** [review]

**Review by grabkowski — DISMISSED** [review]

**Review by edleeks87 — DISMISSED** [review]

File: cypress/legacy-e2e-tests-guide.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+│   └── support/
+│       ├── step-definitions/
+│       │   └── [component-name]-steps.js (files with cucumber steps)
+│       │
+│       └── component-helper/ (contains utilities to help with component tests)
+│
+├── docker-compose.yml
+├── webpack.config.js
+├── tsconfig.json
+└── README.md
```
Comment (DlgSHi) [inline cypress/legacy-e2e-tests-guide.md:43]: Why you have removed the `cypress.config.js` file? :)

```suggestion
└── README.md
└── cypress.config.js
```

Comment (Parsium) [reply cypress/legacy-e2e-tests-guide.md:43]: Looks like I got my wires crossed there, thanks for spotting this @DlgSHi 👍🏼

File: cypress/legacy-e2e-tests-guide.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+│   │       ├── index.js (exported arrow functions for locators)
+│   │       └── locators.js (string const locators)
+│   │
+│   └── support/
+│       ├── step-definitions/
+│       │   └── [component-name]-steps.js (files with cucumber steps)
+│       │
+│       └── component-helper/ (contains utilities to help with component tests)
+│
+├── cypress.config.js
```
Comment (edleeks87) [inline cypress/legacy-e2e-tests-guide.md:40]: Was `README` intentionally taken out?

Comment (DlgSHi) [reply cypress/legacy-e2e-tests-guide.md:40]: we need both, `readme` and `cypress.config.js` files

**Review by DlgSHi — DISMISSED** [review]

**Review by edleeks87 — DISMISSED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Pull request merged**